docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+
+  Adding or removing an exclude-subnet should not introduce network disruption whatsoever, as this setting effects OVN-Kubernetes IPAM service
+  and not the pods network-stack directly.
+
+  In a scenario the external network introduce new "service" whose IP address now considered reserved, e.g.: 10.0.0.10.
+  In case there the address 10.0.0.10 already assigned a pod, both the "service" and the pod may suffer from network disruption  
+  due to IP address conflict.
+  Once the pod with the assigned IP is re-created, it will get different, non excluded, IP address from the pool, and the IP conflict
+  should be resolved.
+
+  Mitigation for this can be providing indication about this state - excludeSubnet cannot be set because IP in its range already
+  been assigned to pods.
+
+
+Once all the connected workloads consume the new network-config, no network disruption should occur as all connected workloads
+network settings are aligned, using the latest network-config.
+
+## OVN Kubernetes Version Skew
+
+## Alternatives
+
+### Indicating pod configured with outdated network-config
+Following discussions about the proposed solution for indicating pod configured with outdated network-config,
+some alternatives and improvements were suggested.
+The proposed solution serves as a baseline for most suggested alternatives.
+Most alternative solutions are additions serving as an additional matter for providing similar indication, as the proposed one.
+
+#### 1. Emit events on pods 
+Similar to the idea of the proposed solution, but with emitting events on the connected pods.
+Following CUDN CR localnet spec change, the CUDN controller should emit event on connected pods.
+
+The event should indicate pod network configuration is outdated, and suggest restarting to consume the latest network config.
+
+Comparing to annotation, events are harder to manage and get garbage collected.
+In addition, filtering events to realize which pod run outdated network-config is not intuitive 
+comparing to filtering by an annotation.
+
+Emitting events is a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 2. Add condition to pod
+Similar to the idea of the proposed solution and option (1), but with adding status condition to the connected pods.
+
+Following CUDN CR localnet spec change, the CUDN controller should add a pod condition on the connected pods,
+providing indication about OVN-K network configuration state.
+The pod condition should be `true` when the pod network-config is up-to-date.
+The pod condition should be `false` when the pod network-config is outdated.
+
+Adding condition to pods is a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 3. CUDN CRD status condition
```

Comment:
I think this is also required IMO. @jcaamano fyi. If a user updates the CRD and it causes either some incompatibility such as: excluding an IP that is already reserved, or some other failure that is caused by the update. Then the CUDN should go into error state. 

If the update to the CRD is valid, but for some reason a node's ovnkube-controller detects some error configuring it, then it should post a substatus failure. The overall CUDN CRD then should also go into error state. I think @npinaeva already has groundwork/plans to do this?

Reply:
Do you mean the NetworkReady condition we were going to report from every node? 
We could do that, but it requires some work for per-node condition reporting + centralized status, and I think currently ovn-k is totally fine with leaving excludedSubnet IP on the pod if it is already assigned (: so some extra changes to report these errors may be needed

Reply:
I think the first part is covered today, if the CRD spec is invalid or has some bad input, the CRD status will communicate the issue through conditions.
If there was an error with rendering/creating the NAD, will `NetworkCreated` condition will be false.
If there was an error related to the node allocator, `NetworkAllocationSucceeded` condition will be false.